API/Controllers/FilterController.cs

```diff
@@ -120,4 +122,39 @@ public ActionResult<FilterV2Dto> DecodeFilter(DecodeFilterDto dto)
     {
         return Ok(SmartFilterHelper.Decode(dto.EncodedFilter));
     }
+
+    [HttpPost("rename")]
+    public async Task<ActionResult> RenameFilter([FromQuery] int filterId, [FromQuery] string name)
+    {
+        var user = await _unitOfWork.UserRepository.GetUserByIdAsync(User.GetUserId(), AppUserIncludes.SmartFilters);
+        if (user == null) return Unauthorized();
+
+        if (User.IsInRole(PolicyConstants.ReadOnlyRole))
+        {
+            return BadRequest(await _localizationService.Translate(User.GetUserId(), "permission-denied"));
+        }
+
+        if (string.IsNullOrWhiteSpace(name))
+        {
+            return BadRequest("Name must be set");
+        }
+
+        if (Seed.DefaultStreams.Any(s => s.Name.Equals(name, StringComparison.InvariantCultureIgnoreCase)))
+        {
+            return BadRequest("You cannot use the name of a system provided stream");
+        }
+
+        var filter = user.SmartFilters.FirstOrDefault(f => f.Id == filterId);
+        if (filter == null)
+        {
+            return BadRequest(await _localizationService.Translate(User.GetUserId(), "filter-not-found"));
+        }
+
+        filter.Name = name.Trim();
```

Comment:
You are modifying name here but comparing it against the Default Streams, you would need to trim it on the first line.